evm_loader/lib/src/types/tracer_ch_common.rs

```diff
@@ -129,3 +138,65 @@ pub struct EthSyncing {
     pub current_block: u64,
     pub highest_block: u64,
 }
+
+pub struct RevisionMap {
+    map: BTreeMap<u64, String>,
+    pub last_update: Instant,
+}
+
+impl RevisionMap {
+    pub fn new(neon_revision_ranges: Vec<(u64, u64, String)>) -> Self {
+        let mut map = BTreeMap::new();
+
+        for (start, end, value) in neon_revision_ranges {
+            map.insert(start, value.clone());
+            map.insert(end, value);
+        }
+
+        let last_update = std::time::Instant::now();
+
+        RevisionMap { map, last_update }
+    }
+
+    // When deploying a program for the first time it is now only available in the next slot (the slot after the one the deployment transaction landed in).
+    // When undeploying / closing a program the change is visible immediately and the very next instruction even within the transaction can not access it anymore.
+    // When redeploying the program becomes temporarily closed immediately and will reopen with the new version in the next slot.
+    pub fn build_ranges(input: Vec<(u64, String)>) -> Vec<(u64, u64, String)> {
+        let mut ranges = Vec::new();
+
+        if input.is_empty() {
+            return ranges;
+        }
```

Comment:
I think this condition can be safely removed.

Reply:
Yes, you're right.